xmppserver/src/main/java/org/jivesoftware/openfire/net/DNSUtil.java

```diff
@@ -90,28 +91,39 @@ public class DNSUtil {
      *
      * If that lookup fails as well, it's assumed that the XMPP server lives at
      * the host resolved by a DNS A lookup at the specified domain on the
-     * specified default port.<p>
+     * specified default port.
      *
      * As an example, a lookup for "example.com" may return "im.example.com:5269".
      *
+     * The returned collection is a list of sets of host names. The 'inner'
+     * collection, the sets of host names, are grouping host names that have an
+     * equal 'priority' SRV value. The 'outer' collection, the list, defines the
+     * order that's defined by the 'priority' value (lowest first). Generally
+     * speaking, all values from the first set of the outer list should be
+     * processed, before elements from the next set of that list are to be processed.
+     *
+     * The Sets that are returned as elements of the list have a predictable
+     * iteration order. This order is based on a randomization based on the
+     * SRV 'weight' value of each host.
+     *
      * @param domain the domain.
      * @param defaultPort default port to return if the DNS look up fails.
      * @return a list of  HostAddresses, which encompasses the hostname and port
      *         that the XMPP server can be reached at for the specified domain.
      * @see <a href="https://tools.ietf.org/html/rfc6120#section-3.2">XMPP CORE</a>
      * @see <a href="https://xmpp.org/extensions/xep-0368.html">XEP-0368</a>
      */
-    public static List<HostAddress> resolveXMPPDomain(String domain, int defaultPort) {
+    public static List<Set<WeightedHostAddress>> resolveXMPPDomain(String domain, int defaultPort) {
```

Comment:
Given the only place (in this commit - feel free to ignore if this turns out to be wrong in subsequent commits) that this is used, the return value is immediately transformed in to an ordered list, perhaps this method should return that ordered list?

Reply:
I think that's not correct: the return value is used in SocketUtil to process all elements of the first set, before moving to the elements of the next set. 

I think one of my earlier commits didn't modify the SocketUtil usage of this code yet (apart from changes to make the compiler happy). This comment is highlighted by GitHub to be 'outdated', so I'm guessing this comment is based on a since-modified bit of code.

Reply:
Yes, sounds like it :)